code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm

```diff
@@ -97,12 +97,23 @@
 	X.visible_message("<span class='notice'>\The [X] digs itself into the ground!</span>", \
 		"<span class='notice'>You dig yourself into place! If you move, you must wait again to fire.</span>", null, 5)
 	X.set_bombard_pointer()
+	RegisterSignal(X, COMSIG_MOB_ATTACK_RANGED, /datum/action/xeno_action/activable/bombard/proc.on_ranged_attack)
+
 
 /datum/action/xeno_action/activable/bombard/on_deactivation()
 	var/mob/living/carbon/xenomorph/boiler/X = owner
 	if(X.selected_ability == src)
 		X.reset_bombard_pointer()
 		to_chat(X, "<span class='notice'>You relax your stance.</span>")
+	UnregisterSignal(X, COMSIG_MOB_ATTACK_RANGED)
+
+
+/datum/action/xeno_action/activable/bombard/proc/on_ranged_attack(mob/living/carbon/xenomorph/X, atom/A, params)
+    if(!X.selected_ability)
+        return
+    if(X.selected_ability.can_use_ability(A))

```

Comment:
this is just the same shitcode

Reply:
LaK has asked for an example.
If you want to review PRs and for your opinion to be taken into account, then start providing some feedback.